src/modules/Elsa.Workflows.Api/Options/ApiOptions.cs

```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Elsa.Workflows.Api.Options;
+
+/// <summary>
+/// Options for the api module.
+/// </summary>
+public class ApiOptions
+{
+    /// <summary>
+    /// A mode that does not allow editing workflows.
+    /// </summary>
+    public bool IsReadOnlyMode { get; set; }
```

Comment:
I actually put it first in the management layer but then decided that it is rather an Api concern rather than a related to workflow management. For me it feels like the management gives you the possibility for the API and the API decides how it has to use it.

The read-only mode is consumed by the HATOAS and authorization, both being API concerns.

What do you think?

Reply:
I see your point. However, one could argue that the management layer is also an API; just a different audience (C# developers building their own services and controllers around management, bypassing the REST APIs). What would you say to that?

Reply:
Moved it to the management layer.